setup.sh

```diff
@@ -1,10 +1,113 @@
 #!/usr/bin/env bash
 
-export SERVER_ADDRESS="$1"
-export HULY_SECRET="secret"
+HULY_VERSION="v0.6.295"
+DOCKER_NAME="huly"
+CONFIG_FILE="huly.conf"
 
-echo "Setting Huly Server Address: $SERVER_ADDRESS"
-envsubst < template.conf > nginx.conf
-envsubst < template.env > .env
+if [ -f "$CONFIG_FILE" ]; then
+    source "$CONFIG_FILE"
+fi
 
-./use-version.sh v0.6.295
+while true; do
+    if [[ -n "$HOST_ADDRESS" ]]; then
+        prompt_type="current"
+        prompt_value="${HOST_ADDRESS}"
+    else
+        prompt_type="default"
+        prompt_value="<IP>"
+    fi
+    read -p "Enter the host address (domain name or IP) [${prompt_type}: ${prompt_value}]: " input
+    _HOST_ADDRESS="${input:-${HOST_ADDRESS:-$(hostname -i)}}"
+    break
+done
+
+while true; do
+    if [[ -n "$HTTP_PORT" ]]; then
+        prompt_type="current"
+        prompt_value="${HTTP_PORT}" 
+    else
+        prompt_type="default"
+        prompt_value="80"
+    fi
+    read -p "Enter the port for HTTP [${prompt_type}: ${prompt_value}]: " input
+    _HTTP_PORT="${input:-${HTTP_PORT:-80}}"
+    if [[ "$_HTTP_PORT" =~ ^[0-9]+$ && "$_HTTP_PORT" -ge 1 && "$_HTTP_PORT" -le 65535 ]]; then
+        break
+    else
+        echo "Invalid port. Please enter a number between 1 and 65535."
+    fi
+done
+
+if [[ "$_HOST_ADDRESS" == "localhost" || "$_HOST_ADDRESS" =~ ^([0-9]{1,3}\.){3}[0-9]{1,3}$ ]]; then
+    _HOST_ADDRESS="${_HOST_ADDRESS}:\${HTTP_PORT}"
```

Comment:
Did you escape `$` intentionally? For me it sets HOST_ADDRESS to `localhost:${HTTP_PORT}` that looks confusing in the output.

Reply:
Yes, we want the user to only have to change the ip in one place
Incase the user uses an IP as `HOST_ADDRESS` the port has to be added